google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/spi/v1/MessageDispatcher.java

```diff
@@ -475,7 +473,7 @@ private void processOutstandingAckOperations(
       if (!pendingAcks.isEmpty()) {
         try {
           acksToSend = new ArrayList<>(pendingAcks);
-          logger.log(Level.INFO, "Sending {} acks", acksToSend.size());
+          logger.log(Level.INFO, "Sending " + acksToSend.size() + " acks");
```

Comment:
Why are you converting the log into string addition? The string formatting is more performant in the case that logging is not performed because it is higher than the log level (e.g. if it is set to `WARN`). (occurs in multiple places in this PR)